lib/streamlit/dataframe_util.py

```diff
@@ -309,7 +309,7 @@ def convert_anything_to_pandas_df(
     # back to Arrow when marshalled to protobuf, but area/bar/line charts need
     # DataFrame magic to generate the correct output.
     if hasattr(data, "to_pandas"):
-        return cast(pd.DataFrame, data.to_pandas())
+        return pd.DataFrame(data.to_pandas())
```

Comment:
I am curious: should we add a check for `if isinstance(pd.DataFrame, data)` and only create the dataframe in that case and otherwise keep the cast or is this over-optimizing?

Reply:
We could, but it would probably even need to be a bit more complicated since, in some situations, we need to ensure that it's actually a copy. The one-liner already ensures that. Since this `to_pandas` is only meant as a fallback for unknown data structures, we probably don't need to optimize it a lot more. This PR: https://github.com/streamlit/streamlit/pull/9045/files will add dedicated optimized logic for many data formats. And if we see that some data structure is used more and more within our metric data, we probably just add some dedicated optimized logic.

Reply:
tldr: i think for the fix here it is fine as is, but I will think about it for the improvements in https://github.com/streamlit/streamlit/pull/9045/files